wgpu/backends/rs.py

```diff
@@ -115,6 +115,33 @@ def _get_wgpu_lib_path():
 ffi.set_source("wgpu.h", None)
 _lib = ffi.dlopen(_get_wgpu_lib_path())
 
+# Get the actual wgpu-native version
+__version_lib__ = ffi.string(_lib.wgpu_get_version()).decode()
+version_info_lib = tuple(map(int, __version_lib__.split(".")))
+if __version__ != __version_lib__:
+    logger.warn(f"Expected wgpu-native version {__version__} but got {__version_lib__}")
+
+
+@ffi.callback("void(int level, const char *)")
+def _logger_callback(level, c_msg):
+    msg = ffi.string(c_msg).decode(errors="ignore")  # makes a copy
+    m = {
+        1: logger.error,
+        2: logger.warning,
+        3: logger.info,
+        4: logger.debug,
+        5: logger.debug,  # trace
+    }
+
+    func = m.get(level, logger.warn)
+    func(msg)
+
+
+# Set log callback. Levels -> 1: error, 2: warn, 3: info, 4: debug
+_lib.wgpu_set_log_callback(_logger_callback)
+_lib.wgpu_set_log_level(2)
+logger.setLevel(logging.WARN)
+# todo: somehow keep log levels in sync. At least automate log-level mapping
```

Comment:
idea: a function that sets both levels at once? would also make it possible for users to set a different level.

Reply:
Yes, I think that makes the most sense. We can do a separate function. Or we can subclass the logger class and re-implement `setLevel()` to also call `wgpu_set_log_level`. The latter is a bit more complex, but probably cleaner.

Reply:
When subclassing, users can still run `logger.getLogger("wgpu"); logger.setLevel(...)` and they wouldn't get the subclass, right? I'm not sure if there's an airtight solution?

Reply:
Yes, I think that `getLogger()` would return the subclass. So that would make it trivial to keep it in sync, right? 

> [...] what if you just set the rust logger to the lowest level [...] and only filter with the python logger level?

That would cause a lot of overhead. First for the Rust logging system, and then sending them over to Python (turning strings into zero-terminated char arrays) ...

Reply:
> Yes, I think that `getLogger()` would return the subclass. So that would make it trivial to keep it in sync, right?
> 
> > [...] what if you just set the rust logger to the lowest level [...] and only filter with the python logger level?
> 
> That would cause a lot of overhead. First for the Rust logging system, and then sending them over to Python (turning strings into zero-terminated char arrays) ...

Really now! Awesome.

Reply:
```py
import logging

class MyLogger(logging.Logger):
    def setLevel(self, level):
        super().setLevel(level)
        print("Setting log level to ", level)

original_cls = logging.getLoggerClass()
logging.setLoggerClass(MyLogger)
logging.getLogger("wgpu")
logging.setLoggerClass(original_cls)

logger = logging.getLogger("wgpu")
logger.setLevel(logging.INFO)
```

```
>>> (executing cell "" (line 39 of "tmp.py"))
Setting log level to  20
```

Reply:
`setLoggerClass`, I never knew about that part of the logging API. will surely use this in the future